virtual_ecosystem/models/animal/animal_cohorts.py

```diff
@@ -177,21 +238,32 @@ def defecate(
         TODO: Update with stoichiometry
 
         Args:
-            excrement_pool: The local ExcrementSoil pool in which waste is deposited.
+            excrement_pools: The ExcrementPool objects in the cohort's territory in
+                which waste is deposited.
             mass_consumed: The amount of mass flowing through cohort digestion.
         """
-        # Find total waste mass, the total amount of waste is then found by the
-        # average cohort member * number individuals.
-        waste_energy = mass_consumed * self.functional_group.conversion_efficiency
+        # the number of communities over which the feces are to be distributed
+        number_communities = len(excrement_pools)
 
-        # This total waste is then split between decay and scavengeable excrement
-        excrement_pool.scavengeable_energy += (
-            (1 - self.decay_fraction_excrement) * waste_energy * self.individuals
-        )
-        excrement_pool.decomposed_energy += (
-            self.decay_fraction_excrement * waste_energy * self.individuals
+        # Find total waste mass, the total amount of waste is found by the
+        # average cohort member * number individuals.
+        waste_energy = (
+            mass_consumed
+            * self.functional_group.conversion_efficiency
+            * self.individuals
         )
 
+        waste_energy_per_community = waste_energy / number_communities
+
+        for excrement_pool in excrement_pools:
```

Comment:
How many pools (i.e. number of communities) are we talking about? If they are many, having the `excrement_pools` as an array and somewhat vectorising this (or at least, using list comprehensions) might speed up things here compared with a plain loop.